client/src/services/releaseVirtualDocument.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2022 Nicola Dardanis
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode';
+import { FlixRelease } from './releases';
+import { getInstalledFlixVersion } from './state';
+
+/**
+ * The scheme used to distinguish our documents.
+ */
+const scheme = 'flixcompiler';
+
+/**
+ * Handle documents where the uri scheme is [scheme] (flixcompiler):
+ *  - check whether the current installed version of the compiler is the same for which we want to display the release changelog.
+ *  - returns the body of the document to be displayed.
+ */
+const flixReleaseDocumentProvider = new (class implements vscode.TextDocumentContentProvider {
+  provideTextDocumentContent(uri: vscode.Uri): string {
+      // use uri-path as text
+      const id = new Number(uri.path);
+      const installedVersion = getInstalledFlixVersion();
+      if (id == installedVersion.id) {
+        return `Version: ${installedVersion.version.major}.${installedVersion.version.minor}.${installedVersion.version.patch}\n`
+            + `${installedVersion.description}`;
+      } else {
+        throw new Error(`The current installed compiler (${installedVersion.id}) doesn't match the requested one (${id}).`);
```

Comment:
How can this happen? Do we really want to throw here? 

Does this assume the programmer clicks yes to update the compiler version?

Reply:
No it should never happen, it is only possible if someone later will change the place where the function `openFlixReleaseOverview` is invoked, as for now it should always work as expected

Reply:
I just added it to have better error reporting in case in the future this would happen, but I see your point

Reply:
My fear is that some day this breaks and then the whole extensions breaks. In that case it would better just to not show the changes.

What happens if you run and old version and you decline to update? Do you get the exception then?

Reply:
Nothing happens, since this functionality is only triggers when the installed version is actually set, so just when the download happens successfully

Reply:
Also, I have tried to trigger the exception and the extension code it's not interrupted and just continue the normal execution without showing the CHANGELOG. Indeed the code of the provider is not called by us and is internally triggered by the call to `executeCommand`.

Using a `console.error` would instead let the CHANGELOG be displayed incorrectly with an empty body

Reply:
I am still just concerned about this crashing the extension and making it unusable.

What if we just return "Unable to get latest changelog. Please see XYZ" or something like that?